pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -167,34 +178,146 @@ func checkVersion(instance *servingv1alpha1.KnativeServing) error {
+func waitForServiceMeshControlPlaneReady(operatorNamespace string) error {
+	smcp := &maistrav1.ServiceMeshControlPlane{}
+	return wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		if err := api.Get(context.TODO(), client.ObjectKey{Namespace: ingressNamespace(operatorNamespace), Name: smcpName}, smcp); err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		for _, cond := range smcp.Status.Conditions {
+			if cond.Type == maistrav1.ConditionTypeReady && cond.Status == maistrav1.ConditionStatusTrue {
+				return true, nil
 			}
 		}
-		return false
+		return false, nil
+	})
+}
+
+// installServiceMeshControlPlane installs serviceMeshControlPlane
+func installServiceMeshControlPlane(instance *servingv1alpha1.KnativeServing) error {
+	const (
+		path = "deploy/resources/serviceMesh/smcp.yaml"
+	)
+	manifest, err := mf.NewManifest(path, false, api)
+	if err != nil {
+		log.Error(err, "Unable to create serviceMeshControlPlane install manifest")
+		return err
+	}
+	transforms := []mf.Transformer{
+		mf.InjectOwner(instance),
+		mf.InjectNamespace(ingressNamespace(instance.GetNamespace())),
+	}
+	if err := manifest.Transform(transforms...); err != nil {
+		log.Error(err, "Unable to transform serviceMeshControlPlane manifest")
+		return err
+	}
+	if err := manifest.ApplyAll(); err != nil {
+		log.Error(err, "Unable to install serviceMeshControlPlane")
+		return err
+	}
+	return nil
+}
+
+// installServiceMeshMemberRole installs serviceMeshMemberRole for knative-serving namespace
+func installServiceMeshMemberRole(operatorNamespace string) error {
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: ingressNamespace(operatorNamespace), Name: smmrName}, smmr); err != nil {
+		if apierrors.IsNotFound(err) {
+			smmr.Name = smmrName
+			smmr.Namespace = ingressNamespace(operatorNamespace)
+			smmr.Spec.Members = []string{operatorNamespace}
+			return api.Create(context.TODO(), smmr)
+		}
+		return err
 	}
-	if !found(smmr) {
-		return checkError(errors.New("knative-serving namespace is not a configured member in serviceMeshMemberRoll"), instance)
+	var exist = false
+	// If serviceMeshMemberRole already exist than check for knative-serving ns is configured member or not
+	for _, member := range smmr.Status.ConfiguredMembers {
+		if member == operatorNamespace {
+			exist = true
+			break
+		}
+	}
+	// if knative-serving ns is not a configured by any chance than update existing serviceMeshMemberRole
+	if !exist {
+		smmr.Spec.Members = append(smmr.Spec.Members, operatorNamespace)
+		return api.Update(context.TODO(), smmr)
 	}
-	log.Info("All dependencies are installed")
-	instance.Status.MarkDependenciesInstalled()
 	return nil
 }
 
+// waitForServiceMeshMemberRoleReady Checks knative-serving namespace is a configured member or not
+func waitForServiceMeshMemberRoleReady(operatorNamespace string) error {
```

Comment:
Can we remove the name `operatorNamespace` with something like `ksNamespace` or `servingNamespace`? This will not be the location of the operator in a system installed via OLM.